### DISCUSSION TITLE
How to do inferences with a model that is trained with tiling mechanism?

### DISCUSSION THREAD

**Discussion by masaoskari** [open-edge-platform/anomalib#2427] [Q&A] [answered]
Hi, 

I'm confused about how to do inferences for the models that are trained with tiling mechanism. For example if model is trained like this (got this from docs):
```
# Import the required modules
from anomalib.data import MVTec
from anomalib.engine import Engine
from anomalib.models import Padim
from anomalib.callbacks import TilerConfigurationCallback
# Initialize the datamodule and model
datamodule = MVTec(num_workers=0, image_size=(128, 128))
model = Padim()
# prepare tiling configuration callback
tiler_config_callback = TilerConfigurationCallback(enable=True, tile_size=[128, 64], stride=64)
# pass the tiling configuration callback to the engine
engine = Engine(image_metrics=["AUROC"], pixel_metrics=["AUROC"], callbacks=[tiler_config_callback])
# train the model (tiling is seamlessly utilized in the background)
engine.fit(datamodule=datamodule, model=model)
````
How then do the inference for the model? Do I need to tile the test image also somehow and feed the model? I guess when tiling is used there are separated model for each tile, so how to use these. Please give concrete example and update docs.

**Comment by blaz-r** [comment, marked as answer]
This is single model tiling config, so the Tiler is part of the model. This means that the image will be tiled in the forwards pass and you don't need to do anything special. I am not sure if this works for onnx though, but should work fine with lightning and torch.

If you want to use the tiled ensemble, where a separate model is trained for each tile refer to [the Tiled ensemble docs](https://anomalib.readthedocs.io/en/latest/markdown/guides/how_to/pipelines/tiled_ensemble.html). In that case you have a pipeline for ensemble training and a pipeline for evaluation (prediction).

**Reply by masaoskari** [reply]
Thanks! So had a bit miss understood the things, but now all good 💯
